dataland-backend/src/main/kotlin/org/dataland/datalandbackend/controller/DataController.kt

```diff
@@ -76,17 +91,18 @@ abstract class DataController<T>(
     }
 
     override fun getCompanyAssociatedData(dataId: String): ResponseEntity<CompanyAssociatedData<T>> {
-        val companyId = dataMetaInformationManager.getDataMetaInformationByDataId(dataId).company.companyId
+        val metaInfo = dataMetaInformationManager.getDataMetaInformationByDataId(dataId)
+        if (!isDataViewableByUser(metaInfo, DatalandAuthentication.fromContextOrNull())) {
```

Comment:
warum fromContextOrNull?
warum nicht fromContext?

Reply:
Because it is also called from the sample company view page where no one is logged in. But let me check once more if we use some kind of authentication though.

Reply:
When the sample page requests data it has no authentication.

Reply:
ok